<?xml version="1.0"?>

<launch>

<!-- <include file="$(find service_utd)/launch/quadrotor_empty_world.launch" /> -->
<param name="/use_sim_time" value="true" />
<param name="timefactor" value="5" />
<node name="clockserver" pkg="service_utd" type="clockserver" output="screen" required="true" />

<!--group ns="uav1">
<include file="$(find service_utd)/launch/spawn_quadrotor_with_downward_cam.launch">
  <arg name="name" value="uav1" />
  <arg name="tf_prefix" value="uav1" />
  <arg name="x" value="-1.0" />
  <arg name="y" value="24.362551" />
  <arg name="z" value="0.0" />
</include>
</group>

<group ns="uav2">
<include file="$(find service_utd)/launch/spawn_quadrotor_with_downward_cam.launch">
  <arg name="name" value="uav2" />
  <arg name="tf_prefix" value="uav2" />
  <arg name="x" value="42.280787" />
  <arg name="y" value="38.019486" />
  <arg name="z" value="0.0" />
</include>
</group>

<group ns="uav3">
<include file="$(find service_utd)/launch/spawn_quadrotor_with_downward_cam.launch">
  <arg name="name" value="uav3" />
  <arg name="tf_prefix" value="uav3" />
  <arg name="x" value="41.764476" />
  <arg name="y" value="12.914353" />
  <arg name="z" value="0.0" />
</include>
</group>

<group ns="uav4">
<include file="$(find service_utd)/launch/spawn_quadrotor_with_downward_cam.launch">
  <arg name="name" value="uav4" />
  <arg name="tf_prefix" value="uav4" />
  <arg name="x" value="5.969434" />
  <arg name="y" value="-1.0" />
  <arg name="z" value="0.0" />
</include>
</group-->

<group ns="target">
  <!-- Load the URDF into the ROS Parameter Server -->
	<!-- <param name="robot_description"
		command="$(find xacro)/xacro.py '$(find p3dx_description)/urdf/pioneer3dx.xacro'" /> -->
    <!-- Run a python script to the send a service call to gazebo_ros to spawn
  		a URDF robot -->
  <!--node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model"
  		respawn="false" output="screen" args="-urdf -model p3dx -param robot_description" /-->
  <rosparam param="roverinit">[6.969434,0.0,0,0]</rosparam>
  <node name="fakerover" pkg="service_utd" type="fakerover" output="screen" />
  <param name="mapname" value="$(find service_utd)/matlabmap.yml" />
  <param name="odometry_topic" value="p3dx/base_pose_ground_truth" />
  <node name="rover_controller" pkg="service_utd" type="rover_controller" output="screen">
    <!--remap from="cmd_vel" to="cmd_vel_mux/input/teleop" /-->
  </node>
  <!-- Just in case we want to replace the Pioneer with a Turtlebot -->
  <!--include file="$(find turtlebot_gazebo)/launch/includes/kobuki.launch.xml">
    <arg name="base" value="kobuki"/>
    <arg name="stacks" value="hexagons"/>
    <arg name="3d_sensor" value="kinect"/>
  </include-->

</group>

</launch>
